### DISCUSSION TITLE
how to use two mailbox-sensors

### DISCUSSION THREAD

**Discussion by steuerzentrale** [PricelessToolkit/MailBoxGuard#32] [Q&A]
Hi there

sorry for asking so many questions :P I recently installed a mailbox-sensor and would like to install another one. The mail and parcels are in separated departements inside my mailbox. How can I "pair" both to the Lora-gateway?

Can I use the same arduino-sketch for the second mailbox-sensor and the sensor just appears in HA as a second one? If not, what is the way operate two mailbox-sensors with one gateway?

**environment**
Home Assistant supervised
mailbox-sensor with Auto-MQTT discovery sketch
Lora-gateway with Auto-MQTT discovery sketch

thanks for the clarification

**Comment by PricelessToolkit** [comment]
Hi, you can ask as many questions as you want 😊
I use  "LoRa_Gateway_MQTT_Simple" because it publishes row data "Keys" so I can distinguish which sensor it is. 
PS; you need to use "Mailbox_Guard_Sensor.ino" for each sensor with different keys.

Don't forget to add a manual sensor

```yaml
mqtt:
    sensor:
     - name: "LoRa_Code"
       state_topic: "LoRa-Gateway/Code"

     - name: "LoRa_RSSI"
       state_topic: "LoRa-Gateway/RSSI"

**Comment by steuerzentrale** [comment]
thats cool thanks. so to be sure, I use the LoRa_Gateway_MQTT_Simple and put in the ssid, password, mqtt_username, mqtt_password, mqtt_server and mqtt_port.

on the sensor I use Mailbox_Guard_Sensor.ino I put in the NewMailCode,  LowBatteryCode and define the band. the NewMailCode and LowBatteryCode must be different for both sensors. thats it?

I dont have to change the sync-word or something else than above?

finally i add the manual sensors in HA.

**Comment by PricelessToolkit** [comment]
that's it if it not will be triggered at the same time.

**Reply by danielducu** [reply]
But what about situations where two sensors are triggered at the same time ? Is there a workaround ? Maybe modify gateway to check iterate trough multiple defined sync words?

**Reply by PricelessToolkit** [reply]
if exactly at the same time, even NASA can't handle that.  But if 0.1 seconds apart, you can set SF7, SF8, or SF9 and signals will not jam each other


For example sending  "6 characters" of data using different Spreading Factors (SF), the estimated time on air (ToA)
// for each SF is as follows:

// SF7: Approximately 0.027 seconds (27.494 milliseconds)
// SF8: Approximately 0.052 seconds (52.224 milliseconds)
// SF9: Approximately 0.100 seconds (100.147 milliseconds)
// SF10: Approximately 0.193 seconds (193.413 milliseconds)
// SF11: Approximately 0.385 seconds (385.297 milliseconds)
// SF12: Approximately 0.746 seconds (746.127 milliseconds)

-------------------------------------------------------------------------------
For optimizing the SPREADING_FACTOR (SF) in your network, it's crucial not to default to SF12 aiming for maximum distance without considering its downsides. SF12, while extending range, significantly slows down data transmission. For example, if your furthest sensor is only 100 meters away, opting for SF7 is more efficient. SF7 is faster, taking only 0.027 seconds to transmit "6 bytes", thus consuming less power compared to SF12, which would take 0.75 seconds for the same task. Therefore, it's essential to choose the SF wisely based on your specific needs and understand the trade-offs. Avoid setting SF12 by default without assessing the impact on speed, power consumption, and time on air (ToA) for others.

**Reply by danielducu** [reply]
Good explanation, there is one thing I don't get, lets say I have one sensor that's far enough that i need to use the sf12 and one that's close and I use the sf7 ... what happens with the SF on the gateway... should I set it to the maximum SF of the farthest sensor and lose the fast sf7 of the closer sensor ?

**Reply by PricelessToolkit** [reply]
you can't set different spreading factors, set for the farthest and the same SF for all.

**Reply by danielducu** [reply]
> you can't set different spreading factors, set for the farthest and the same SF for all.

Yeah, make sense. One more thing hehe... i was reading about the syncword... it seams that cannot be used to filter data... (act as a private identify key for my sensors network) it has to be fixed 0x12 or 0x34 for lorawan ... am i wrong?

**Reply by PricelessToolkit** [reply]
sync word acts as a private identify key for the sensor network! A receiver using one sync word will not (mostly) receive packets from a transmitter using a different Syncword. LoRaWAN\TTN uses 0x34

**Reply by danielducu** [reply]
> sync word acts as a private identify key for the sensor network! A receiver using one sync word will not (mostly) receive packets from a transmitter using a different Syncword. LoRaWAN\TTN uses 0x34

So .., we can use anything in the syncword?

**Reply by PricelessToolkit** [reply]
After searching the web for an answer to your question, I've found myself in a state of confusion. Various sources present differing opinions: some indicate that there are only two specific codes—0x12 for private use and 0x34 for LoRaWAN/The Things Network. However, Meshtastic appears to utilize 0x14. To clarify these discrepancies, a thorough review of the datasheets is required.  


Our AI friend says......

Private Sync Word Range: Typically, any hexadecimal value from 0x00 to 0xFF is programmable, but you should avoid common values used by public applications or specified by other custom integrations to reduce conflicts.
Here are a few examples of private network sync word values you might consider:

0x12 (often used for LoRa private, but still quite common)
0x50 to 0x5F
0xA0 to 0xAF
0xC3
0xD7
0xE2

**Reply by danielducu** [reply]
> After searching the web for an answer to your question, I've found myself in a state of confusion. Various sources present differing opinions: some indicate that there are only two specific codes—0x12 for private use and 0x34 for LoRaWAN/The Things Network. However, Meshtastic appears to utilize 0x14. To clarify these discrepancies, a thorough review of the datasheets is required.  
> 
> 
> 
> 
> 
> Our AI friend says......
> 
> 
> 
> Private Sync Word Range: Typically, any hexadecimal value from 0x00 to 0xFF is programmable, but you should avoid common values used by public applications or specified by other custom integrations to reduce conflicts.
> 
> Here are a few examples of private network sync word values you might consider:
> 
> 
> 
> 0x12 (often used for LoRa private, but still quite common)
> 
> 0x50 to 0x5F
> 
> 0xA0 to 0xAF
> 
> 0xC3
> 
> 0xD7
> 
> 0xE2

I guess all I'm left with is check and see the downsides of not using a standard syncword. Thanks a lot for your time !!!!

**Comment by steuerzentrale** [comment]
I configured testwise my existing mailbox-sensor with Mailbox_Guard_Sensor.ino and the gateway with LoRa_Gateway_MQTT_Simple. It works fine so far, whenever the PIR detects motion, I get the code 0xA2B2. This code stays until I publish a message to LoRa-Gateway/Code, for example 0xLBAT.  When the mailbox-sensor starts from scratch (removing and adding battery) the code is 0xLBAT. 

question: 0xLBAT means low battery. How is that being measured? The battery is fully charged and I also replaced it with another fully charged 14500 3.7V battery.

Is it possible to implement the battery in percent as it is with the Auto-MQTT sketches? My goal is to know the batterys state in order to know when I have to replace or charge it.

**Reply by PricelessToolkit** [reply]
it will send 0xLBAT only when you need to recharge battery 3.3v ~, and when you put the new battery.
